src/result.rs

```diff
@@ -1,9 +1,74 @@
 use std::convert::From;
-use std::fmt::{Debug, Error};
+use std::fmt::{Debug, Display, Error};
 use std::{fmt, io};
 
 use thiserror::Error;
 
+/// Represents the position, as line and column, within a text format ion document.
+#[derive(Clone, Debug)]
+pub enum TextPosition {
+    /// No text position available; implies binary format is being read.
+    None,
+    /// The position represented as line and column.
+    LineColumn(usize, usize),
```

Comment:
Subjective: I have a mild preference for `LineAndColumn` over `LineColumn`.

Reply:
Me too, updated in latest.